popde/adaptive_kde.py

```diff
@@ -117,6 +117,100 @@ def set_adaptive_parameter(self, new_alpha, new_global_bw):
         self.set_alpha(new_alpha)
 
 
+class KDEOptimization(AdaptiveBwKDE):
+    """
+    Optimize bandwidth and alpha by grid search using
+    cross validation with a log likelihood figure of merit 
+    """
+    def __init__(self, data, bandwidth_options, alpha_options, weights=None, input_transf=None, stdize=False, rescale=None, backend='KDEpy', bandwidth=1.0, alpha=0.0, dim_names=None, do_fit=True, n_splits=2):
```

Comment:
I don't think we want `do_fit=True` here, since this would set up the KDE with bw=1 and alpha=0.  Ie we want to initialize all the KDE data but not actually train/fit the KDE until the optimization step.

Reply:
removed here in super_init_. need a review

Reply:
I meant that we should have do_fit=False ... have fixed it in an edit myself

Reply:
ok